### PR TITLE
fix: #tno-2222 - fixed translation of stored state for search filter actions

### PR DESCRIPTION
**Pull request by kylgarmor**
- introduced a more generic way of translating between a boolean and it's presence in an array of Actions
- updated npm tno-core to 0.1.18

Did some housekeeping in an effort to move away from having to have a bunch of booleans on the interface which we then have to make sure are translated when storing on back-end

### CODE REVIEW

File: app/editor/src/features/content/constants/ShowOnlyValues.ts [hunk 1]
```diff
@@ -1,10 +1,7 @@
 export enum ShowOnlyValues {
   PrintContent = 'Print Content',
   HasTopic = 'Has Topic',
-  Commentary = 'Commentary',
-  TopStory = 'Top Story',
   OnTicker = 'On Ticker',
-  Homepage = 'Homepage',
```
Comment (kylgarmor) [inline app/editor/src/features/content/constants/ShowOnlyValues.ts:7]: moved these to a enum more closing associated with the topic of `ActionName`

File: app/editor/src/features/content/constants/showOnlyOptions.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
-import { OptionItem } from 'tno-core';
+import { ActionName, OptionItem } from 'tno-core';
 
 import { ShowOnlyValues } from './ShowOnlyValues';
 
 export const showOnlyOptions = [
   new OptionItem(ShowOnlyValues.PrintContent, ShowOnlyValues.PrintContent),
   new OptionItem(ShowOnlyValues.HasTopic, ShowOnlyValues.HasTopic),
-  new OptionItem(ShowOnlyValues.Commentary, ShowOnlyValues.Commentary),
-  new OptionItem(ShowOnlyValues.TopStory, ShowOnlyValues.TopStory),
```
Comment (kylgarmor) [inline app/editor/src/features/content/constants/showOnlyOptions.ts:9]: replace with usage of new `ActionName` enum

File: app/editor/src/features/content/list-view/components/tool-bar/filter/ShowOnlySection.tsx [hunk 3]
```diff
@@ -110,8 +111,8 @@ export const ShowOnlySection: React.FC<IShowOnlySectionProps> = () => {
                         : [...filter.contentTypes]
                       : filter.contentTypes.filter((x) => x !== ContentTypeName.PrintContent),
                     hasTopic: values.some((o) => o.value === ShowOnlyValues.HasTopic),
-                    commentary: values.some((o) => o.value === ShowOnlyValues.Commentary),
-                    topStory: values.some((o) => o.value === ShowOnlyValues.TopStory),
```
Comment (kylgarmor) [inline app/editor/src/features/content/list-view/components/tool-bar/filter/ShowOnlySection.tsx:114]: replace with usage of new `ActionName` enum

File: app/subscriber/src/features/search-page/utils/getFilterActions.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { ActionName, IActionModel, IFilterActionSettingsModel } from 'tno-core';
+
```
Comment (kylgarmor) [inline app/subscriber/src/features/search-page/utils/getFilterActions.ts:2]: Util method to retrieve what would be the stored state of an action when utilized.  Avoids duplication when creating an action to store

File: libs/npm/core/src/hooks/api/interfaces/IFilterSettingsModel.ts [hunk 5]
```diff
@@ -8,7 +8,6 @@ export interface IFilterSettingsModel {
   size: number;
   defaultSearchOperator?: 'and' | 'or';
   boldKeywords?: boolean;
-  commentary?: boolean;
```
Comment (kylgarmor) [inline libs/npm/core/src/hooks/api/interfaces/IFilterSettingsModel.ts:11]: removed in favor of using `Actions` - avoids unnecessary translation

File: libs/npm/core/src/hooks/api/interfaces/IFilterSettingsModel.ts [hunk 6]
```diff
@@ -35,7 +34,6 @@ export interface IFilterSettingsModel {
   tags?: string[];
   startDate?: string;
   status?: ContentStatusName;
-  topStory?: boolean;
```
Comment (kylgarmor) [inline libs/npm/core/src/hooks/api/interfaces/IFilterSettingsModel.ts:38]: removed in favor of using `Actions` - avoids unnecessary translation

File: app/subscriber/src/features/search-page/utils/filterFormat.ts [hunk 7]
```diff
@@ -32,7 +30,6 @@ export const filterFormat = (filter: IFilterSettingsModel, actions?: IActionMode
     sourceIds: filter.sourceIds ?? [],
     startDate: !!filter.startDate ? filter.startDate : undefined,
     tags: filter.tags ?? [],
-    topStory: filter.topStory ?? false,
```
Comment (kylgarmor) [inline app/subscriber/src/features/search-page/utils/filterFormat.ts:35]: removed in favor of using `Actions` - avoids unnecessary translation

File: app/subscriber/src/features/search-page/utils/filterFormat.ts [hunk 8]
```diff
@@ -1,11 +1,9 @@
 import moment from 'moment';
 import { IActionModel, IFilterSettingsModel } from 'tno-core';
 
-import { getActionFilters } from './getActionFilter';
-
 export const filterFormat = (filter: IFilterSettingsModel, actions?: IActionModel[]) => {
   const settings: IFilterSettingsModel = {
-    actions: getActionFilters(filter, actions ?? []),
```
Comment (kylgarmor) [inline app/subscriber/src/features/search-page/utils/filterFormat.ts:8]: removed as unnecessary

File: app/subscriber/src/features/search-page/components/advanced-search/utils/queryToState.ts [hunk 9]
```diff
@@ -18,7 +18,7 @@ export const queryToState = (queryString: string) => {
     inStory: search.inStory === 'true',
     inByline: search.inByline === 'true',
     frontPage: search.mediaTypeIds?.includes(11) || false,
-    topStory: search.actions?.includes('Top Story') || false,
```
Comment (kylgarmor) [inline app/subscriber/src/features/search-page/components/advanced-search/utils/queryToState.ts:21]: replace inline string with enum usage

File: app/subscriber/src/features/search-page/components/advanced-search/components/sections/MoreOptions.tsx [hunk 10]
```diff
@@ -42,9 +56,15 @@ export const MoreOptions: React.FC = () => {
           <Checkbox
             label="are marked as top stories"
             name="topStory"
-            checked={Boolean(filterSettings.topStory)}
+            checked={filterSettings.actions?.map((a) => a.id).includes(topStoryAction.id)}
             onChange={(e) => {
-              storeSearchFilter({ ...filterSettings, topStory: e.target.checked });
+              const newFilterSettings = {
```
Comment (kylgarmor) [inline app/subscriber/src/features/search-page/components/advanced-search/components/sections/MoreOptions.tsx:61]: replaced usage of boolean with updating the `actions` array

File: app/subscriber/src/features/search-page/components/advanced-search/components/sections/MoreOptions.tsx [hunk 11]
```diff
@@ -42,9 +56,15 @@ export const MoreOptions: React.FC = () => {
           <Checkbox
             label="are marked as top stories"
             name="topStory"
-            checked={Boolean(filterSettings.topStory)}
+            checked={filterSettings.actions?.map((a) => a.id).includes(topStoryAction.id)}
```
Comment (kylgarmor) [inline app/subscriber/src/features/search-page/components/advanced-search/components/sections/MoreOptions.tsx:59]: no more boolean...

**Review by Fosol — APPROVED** [review]
I like it

**Pull request merged**